pyVmomi/SoapAdapter.py

```diff
@@ -1321,14 +1345,20 @@ def InvokeMethod(self, mo, info, args, outerStub=None):
       req = self.SerializeRequest(mo, info, args)
       for modifier in self.requestModifierList:
          req = modifier(req)
-      conn = self.GetConnection()
-      try:
-         conn.request('POST', self.path, req, headers)
-         resp = conn.getresponse()
-      except (socket.error, http_client.HTTPException):
-         # The server is probably sick, drop all of the cached connections.
-         self.DropConnections()
-         raise
+
+      max_retries = 100
```

Comment:
I am trying to understand why can't we just loop 100 times and break if the conn is not None? That way we can avoid these variables(num_retries, retry, fail_next, max_tries), right?